src/math/Math.sol

```diff
@@ -28,4 +28,16 @@ library Math {
             }
         }
     }
+
+    /// @dev Returns x / y rounded up ((x + y - 1) / y).
+    function divUp(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        assembly {
+            z := add(x, y)
+            // Revert if y = 0 or if x + y < x
+            if or(iszero(y), lt(z, x)) {
+                revert(0, 0)
+            }
+            z := div(sub(z, 1), y)
+        }
+    }
```

Comment:
Currently, the MathRef divUp reverts even on theoretically possible 256 bits operations like divUp(MAX_UINT256, 10).
If we want to extend the definition of divUp to every operation possible - reverting only on division by 0 - it is possible and cheaper.
The question is therefore if we want to replicate Aave's behaviour or create a more general function.
Tests must be modified (expect revert only when y = 0 / impossible to compare to MathRef on new scenarios).
```suggestion
    /// @dev Returns x / y rounded up.
    function divUp(uint256 x, uint256 y) internal pure returns (uint256 z) {
        assembly {
            // Revert if y = 0
            if iszero(y){
                revert(0,0)
            }
                
            if gt(mod(x, y), 0) {
                z := 1
            }
            z := add(z,div(x, y))

        }
    }
```
We must demonstrate that z doesn't overflow. 
If x % y = 0, z = 0 + x/y <= x <= uint_max. 
Else, x % y != 0 so x = ky + p with 0 < p < y. z= 1 + x//y = 1 + k overflow only if k >= UINT_MAX. 
But if k >= UINT_MAX, p > 0, y > 0 => x = ky + p > UINT_MAX which is not possible.

Gas reports show use of 444 gas when x is a multiple of y and 449 when it's not. 
We can assume that in the general case, y doesn't divide x and so previous commit (448) might be slightly better in the general case.
Avg gas cost over x knowing y is indeed only better when y < 6
Let's try to negate the second condition: 
```suggestion
    /// @dev Returns x / y rounded up.
    function divUp(uint256 x, uint256 y) internal pure returns (uint256 z) {
        assembly {
            // Revert if y = 0
            if iszero(y){
                revert(0,0)
            }
        
            z := 1
            if iszero(mod(x, y)) {
                z := 0
            }

            z := add(z,div(x, y))

        }
    }
```
Obviously, the demonstration still holds.
Gas reports show use of 451 (+7) gas when x is a multiple of y and 446 (-3) when it's not.
Avg gas cost over x knowing y is better than previous commit as soon as y > 2.
Again, overall behaviour changes here which might be unwanted.

Reply:
This function is not part of Aave but our code, so it's more our choice. Not sure it's safe to do it.

Reply:
I agree with @AtomicAzzaz

Reply:
Wait those functions do not behave the same as the original function, even when there is no overflow. For `x == 0` this returns `z >= 1`. This introduces could introduce small borrows for someone who is a pure suppliers, making the complexity of the future functions significantly higher

Reply:
Wait those functions do not behave the same as the original function, even when there is no overflow. For x == 0 this returns z >= 1. This introduces could introduce small borrows for someone who is a pure suppliers, making the complexity of the future functions significantly higher

Reply:
My bad, this is fine actually. Let's do it !